Sources/Commands/Utilities/SymbolGraphExtract.swift

```diff
@@ -82,7 +82,22 @@ public struct SymbolGraphExtract {
         if includeSPISymbols {
             commandLine += ["-include-spi-symbols"]
         }
-        
+
+        // If this target or any dependencies is is built with C++ interop
+        // enabled then swift-symbolgraph-extract must also enable C++ interop.
+        var cxxInterop = module.cxxInterop()
+        if !cxxInterop {
+            for module in try module.recursiveTargetDependencies() {
```

Comment:
Would it make sense to refactor this into a self contained function that tells you whether or not we need the interop flag?

Reply:
Yeah probably, Im struggling to figure out exactly where to put this logic/the logic egor mentioned above so its accessible from both DumpSymbolGraph and SwiftTargetBuildDescription.